torch/optim/adamax.py

```diff
@@ -228,7 +235,10 @@ def _single_tensor_adamax(params: List[Tensor],
             exp_inf.mul_(beta2).unsqueeze(0),
             grad.abs().add_(eps).unsqueeze_(0)
         ], 0)
-        torch.amax(norm_buf, 0, keepdim=False, out=exp_inf)
+        if not differentiable:
+            torch.amax(norm_buf, 0, keepdim=False, out=exp_inf)
+        else:
+            exp_inf = torch.amax(norm_buf, 0, keepdim=False)
```

Comment:
Is the distinction here to maintain functional-ism/no mutations?

Reply:
This is to avoid an in-place modification of `exp_inf` that will trigger autograd errors :)